Tests/Tests.Shared/ConfigurationTests.cs

```diff
@@ -126,23 +143,15 @@ public void UnableToOpenWithNoKey()
             ReliesOnEncryption();
 
             // Arrange
-            var config = new RealmConfiguration("UnableToOpenWithNoKey.realm");
-            Realm.DeleteRealm(config);  // ensure guarded from prev tests
-            var emptyKey = new byte[64];
-            config.EncryptionKey = emptyKey;
-            using (Realm.GetInstance(config))
+            _configuration.EncryptionKey = new byte[64];
```

Comment:
I guess there are quite a few lines of code we'll have to change to adopt for immutable configurations. While I don't object to the idea, I feel that is best addressed when we make the change (primarily because we don't have a clone method yet).